ambry-router/src/main/java/com.github.ambry.router/GetManager.java

```diff
@@ -38,29 +78,101 @@
+   * @param callback the {@link Callback} object to be called on completion of the operation.
+   */
   void submitGetBlobOperation(String blobId, FutureResult<ReadableStreamChannel> futureResult,
       Callback<ReadableStreamChannel> callback) {
-    //@todo
+    // @todo
   }
 
-  void submitGetBlobInfoOperation(String blobId, FutureResult<BlobInfo> futureResult,
-      Callback<BlobInfo> callback) {
-    //@todo
+  /**
+   * Creates and returns requests in the form of {@link RequestInfo} to be sent to data nodes in order to complete
+   * get operations. Since this is the only method guaranteed to be called periodically by the RequestResponseHandler
+   * thread in the {@link NonBlockingRouter} ({@link #handleResponse} gets called only if a
+   * response is received for a get operation), any error handling or operation completion and cleanup also usually
+   * gets done in the context of this method.
+   * @param requestListToFill list to be filled with the requests created
+   */
+  void poll(List<RequestInfo> requestListToFill) {
+    requestRegistrationCallback.requestListToFill = requestListToFill;
+    for (GetOperation op : getOperations) {
+      op.poll(requestRegistrationCallback);
+      if (op.isOperationComplete() && getOperations.remove(op)) {
+        // In order to ensure that an operation is completed only once, call onComplete() only at the place where the
+        // operation actually gets removed from the set of operations. See comment within close().
+        onComplete(op);
+      }
+    }
   }
 
-  void poll(List<RequestInfo> requests) {
-    //@todo
+  /**
+   * Called for a {@link GetOperation} when the operation is complete. Any cleanup that the GetManager needs to do
+   * with respect to this operation will have to be done here. The GetManager also finishes the operation by
+   * performing the callback and notification.
+   * @param op the {@link PutOperation} that has completed.
+   */
+  void onComplete(GetOperation op) {
+    NonBlockingRouter
+        .completeOperation(op.getFuture(), op.getCallback(), op.getOperationResult(), op.getOperationException());
   }
 
+  /**
+   * Hands over the response to the associated GetOperation that issued the request.
+   * @param responseInfo the {@link ResponseInfo} containing the response.
+   */
   void handleResponse(ResponseInfo responseInfo) {
-    // @todo
+    GetRequest getRequest = (GetRequest) responseInfo.getRequest();
+    GetOperation getOperation = correlationIdToGetOperation.remove(getRequest.getCorrelationId());
+    if (getOperations.contains(getOperation)) {
+      getOperation.handleResponse(responseInfo);
+      if (getOperation.isOperationComplete() && getOperations.remove(getOperation)) {
+        onComplete(getOperation);
+      }
+    }
   }
 
+  /**
+   * Close the GetManager.
+   * Complete all existing get operations.
+   */
   void close() {
-    // @todo
+    for (GetOperation op : getOperations) {
```

Comment:
Is `isOpen.compareAndSet(true, false)` needed to ensure a `GetManager` can be closed for only once? (a duplicate question as raised up in `DeleteOperation`).